### PR TITLE
fix HelpError handling

### PR DESCRIPTION
**Pull request by boneskull**
When a HelpError is thrown (e.g., unknown command, no command specified),
`parse()` and `parseAsync()` now catch it and handle gracefully:

- Error message is printed to stderr
- Help text is displayed to stderr
- `process.exitCode` is set to 1 (no `process.exit()` call)
- Returns result with `helpShown: true` flag

This prevents HelpError from bubbling up to global exception handlers
while still providing useful feedback to the user.

Refactors use of `process.exit()` and documents exiting behavior.

### CODE REVIEW

File: src/bargs.ts [hunk 1]
```diff
@@ -760,22 +760,42 @@ const createCliBuilder = <V, P extends readonly unknown[]>(
...
+    ): Promise<ParseResult<V, P> & { command?: string; helpShown?: boolean }> {
+      try {
+        return (await parseCore(state, args, true)) as ParseResult<V, P> & {
+          command?: string;
+        };
+      } catch (error) {
+        if (error instanceof HelpError) {
+          return handleHelpError(error, state) as ParseResult<V, P> & {
+            command?: string;
+            helpShown: true;
```
[the remark this answers is not in the document]
Comment (boneskull) [reply src/bargs.ts]: Fixed in 33249a7. `__parseWithParentGlobals()` now catches `HelpError` so nested builders render their own help.

File: test/bargs.test.ts [hunk 2]
```diff
@@ -786,23 +800,108 @@ describe('merge() edge cases', () => {
...
-        'build',
-        handle(opt.options({}), () => {}),
-      );
-    // No defaultCommand set
+  describe('HelpError handling', () => {
+    it('catches HelpError on no command, displays help, and sets exitCode', async () => {
+      const originalExitCode = process.exitCode;
+      const stderrWrites: string[] = [];
+      const originalStderrWrite = process.stderr.write.bind(process.stderr);
+
```
[the remark this answers is not in the document]
Comment (boneskull) [reply test/bargs.test.ts]: Fixed in 33249a7. Extracted `withCapturedStderr()` helper that captures stderr output and exitCode.

File: test/parser-commands.test.ts [hunk 3]
```diff
@@ -96,22 +96,36 @@ describe('command parsing', () => {
...
-      cli.parseAsync(['unknown']),
-      'to reject with error satisfying',
-      /Unknown command: unknown/,
-    );
+  it('handles unknown command by showing help and setting exitCode', async () => {
+    const originalExitCode = process.exitCode;
+    const stderrWrites: string[] = [];
+    const originalStderrWrite = process.stderr.write.bind(process.stderr);
+
+    process.stderr.write = ((chunk: unknown) => {
```
[the remark this answers is not in the document]
Comment (boneskull) [reply test/parser-commands.test.ts]: Fixed in 33249a7. Extracted the same `withCapturedStderr()` helper here as well.

File: src/bargs.ts [hunk 4]
```diff
@@ -799,7 +819,22 @@ const parseCore = (
...
+   * @function
+   */
+  const earlyExit = (
+    exitCode: number,
+  ): ParseResult<unknown, readonly unknown[]> & {
+    command?: string;
+    helpShown: true;
+  } => {
+    process.exitCode = exitCode;
+    return { command: undefined, helpShown: true, positionals: [], values: {} };
```
[the remark this answers is not in the document]
Comment (boneskull) [reply src/bargs.ts]: Fixed in 33249a7. Renamed `helpShown` to `earlyExit` since it's set for help, version, and completion output.

File: test/bargs.test.ts [hunk 5]
```diff
@@ -1,14 +1,100 @@
...
+        }))
+        .catch(handleMockExit)
+        .finally(cleanup);
+    }
+    cleanup();
+    return Promise.resolve({
+      exitCode: 0,
+      output: stderrWrites.join(''),
+      result: maybePromise,
+    });
```
[the remark this answers is not in the document]
Comment (boneskull) [reply test/bargs.test.ts]: Fixed in 896f965. Extracted to `test/helpers/mock-exit.ts` and updated to use `Promise.resolve().then(fn)` for proper thenable handling.

File: test/bargs.test.ts [hunk 6]
```diff
@@ -1,14 +1,100 @@
...
 import type { StringOption } from '../src/types.js';
 
 import { bargs, handle, map, merge } from '../src/bargs.js';
 import { opt, pos } from '../src/opt.js';
 
+/**
+ * Custom error thrown when process.exit is mocked and called.
+ */
+class MockExitError extends Error {
+  readonly exitCode: number;
```
[the remark this answers is not in the document]
Comment (boneskull) [reply test/bargs.test.ts]: Fixed in 896f965. Extracted to shared helper at `test/helpers/mock-exit.ts`.

File: test/parser-commands.test.ts [hunk 7]
```diff
@@ -4,12 +4,98 @@
...
+import { expect } from 'bupkis';
 import { describe, it } from 'node:test';
 
 import { bargs, handle } from '../src/bargs.js';
 import { opt, pos } from '../src/opt.js';
 
+/**
+ * Custom error thrown when process.exit is mocked and called.
+ */
+class MockExitError extends Error {
```
[the remark this answers is not in the document]
Comment (boneskull) [reply test/parser-commands.test.ts]: Fixed in 896f965. Same shared helper at `test/helpers/mock-exit.ts`.

File: test/parser-commands.test.ts [hunk 8]
```diff
@@ -4,12 +4,98 @@
...
+    cleanup();
+    return Promise.resolve({
+      exitCode: 0,
+      output: stderrWrites.join(''),
+      result: maybePromise,
+    });
+  } catch (error) {
+    cleanup();
+    return Promise.resolve(handleMockExit(error));
+  }
```
[the remark this answers is not in the document]
Comment (boneskull) [reply test/parser-commands.test.ts]: Fixed in 896f965. Same fix in the shared helper - uses `Promise.resolve().then(fn)` now.

File: src/bargs.ts [hunk 9]
```diff
@@ -1053,6 +1097,38 @@ const generateHelpNew = (state: InternalCliState, theme: Theme): string => {
...
+ * @function
+ */
+/**
+ * Handles HelpError by displaying the error message, showing help, and
+ * terminating the process with exit code 1. This is standard CLI behavior -
+ * when a user provides an unknown command or forgets to specify a required
+ * command, they see help and the process exits.
+ *
+ * @function
+ */
```
[the remark this answers is not in the document]
Comment (boneskull) [reply src/bargs.ts:1118]: Fixed in 896f965. Consolidated into single JSDoc block with accurate description.

File: README.md [hunk 10]
```diff
@@ -707,17 +707,46 @@ See `examples/completion.ts` for a complete example.
...
+  throw new Error('process.exit called');
+}) as typeof process.exit;
+
+try {
+  cli.parse(['--help']);
+} catch {
+  // process.exit was called
+}
+
+process.exit = originalExit;
```
[the remark this answers is not in the document]
Comment (boneskull) [reply README.md]: Fixed in 896f965. Updated example to use `ProcessExitError` sentinel and `finally` block for guaranteed cleanup.
